### PR TITLE
Update Java Errors

### PR DESCRIPTION
**Pull request by SSwiniarski**
### Description

Added NullPointerException to the list of example errors. From the Target Keyword list.

Also, not done here, but suggested in the Keyword list, is to have term entries for Java errors in place of (or in addition to) the examples given. How does everyone feel about that?

### Type of Change

- Editing an existing entry (fixing a typo, bug, issues, etc)

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/java/concepts/errors/errors.md
Comment (yangc95) [inline content/java/concepts/errors/errors.md:30]:
```suggestion
- The `NullPointerException`, which occurs when attempting to use a `null` value in place of where an object is required.
```

Comment (yangc95) [reply content/java/concepts/errors/errors.md:30]: I think this is good with just one review. Will merge after

**Comment by SSwiniarski** [comment]
@yangc95, I implemented your change. Good to go.

**Comment by SSwiniarski** [comment]
Also, @yangc95, @Dusch4593, @KTom101: how do you feel about the suggestion of breaking out error entries into their own term entries under error? I was suggested in the Keyword list.

**Review by yangc95 — APPROVED** [review]

**Comment by yangc95** [comment]
> Also, @yangc95, @Dusch4593, @KTom101: how do you feel about the suggestion of breaking out error entries into their own term entries under error? I was suggested in the Keyword list.

If you mean linking each error entry to the General [error](https://www.codecademy.com/resources/docs/general/error) entry, then I'm all for it 👍

**Comment by KTom101** [comment]
> Also, @yangc95, @Dusch4593, @KTom101: how do you feel about the suggestion of breaking out error entries into their own term entries under error? I was suggested in the Keyword list.

I agree with @yangc95. It would be nice to be able to have access to all the error entries under the general error entry. I'm also for having the information in each language's entry as term entries. Especially for the Java one.

**Review by KTom101 — APPROVED** [review]

**Comment by yangc95** [comment]
@SSwiniarski did you want to make these changes on this PR, in which case we can hold off on merging this byte-sized change. Lmk!

**Comment by SSwiniarski** [comment]
@yangc95, there seems to be a consensus. I'll work on updating this PR with term entries for the individual Java errors, and link them back to the General error entry. I'll tag everyone when I'm done.

**Comment by SSwiniarski** [comment]
@yangc95, @Dusch4593, @KTom101: I updated this entry to have term entries for each error.

**Review by yangc95 — COMMENTED** [review]
A few things I noticed! @KTom101 is there anything else you have in mind for improvement?

File: content/java/concepts/errors/errors.md
Comment (Dusch4593) [inline content/java/concepts/errors/errors.md:21]: Given that this PR will merge new term entries for the "Java: Errors" (and thus create a new terms table on the bottom by default) do we really need a list here that might be repeating information?

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@SSwiniarski Left a few last comments/suggestions for you but I did have one sort-of general comment/request: can we include a brief "Examples" section for each term entry? Just so a reader knows what the code involved with these errors might look like?

File: content/java/concepts/errors/errors.md
Comment (SSwiniarski) [reply content/java/concepts/errors/errors.md:23]: I think it might be good, just to make the distinction between errors and exceptions, that the term list isn't going to do for us. And eventually there will be a lot more terms than we have in these lists.

**Comment by SSwiniarski** [comment]
@Dusch4593, I'll look at updating these with some examples.

File: content/java/concepts/errors/errors.md
Comment (Dusch4593) [reply content/java/concepts/errors/errors.md:23]: Yeah that is an unfortunate side-effect of the terms tables. Let's go with what we have right now and see how things grow from here? (Maybe engineering can address that issue with term table organization in a future version).

**Review by Dusch4593 — APPROVED** [review]
@SSwiniarski LGTM! 👍

**Pull request merged**
